src/compiler/emitter.ts

```diff
@@ -345,6 +345,15 @@ var __extends = (this && this.__extends) || function (d, b) {
     d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
 };`;
 
+        const assignHelper = `
+var __assign = (this && this.__assign) || Object.assign || function(t) {

```

Comment:
Polyfill I found (and what React used) convert `t` into an object, like:

```
if (t == null) {
throw new TypeError('Object.assign cannot be called with null or undefined');
}
var result = Object(t);
```

I'm unsure if this is necessary with current browsers. But https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Polyfill does to.